lib/sandbox/execute-context.js

```diff
@@ -52,21 +44,36 @@ module.exports = function (scope, code, execution, console, timers, pmapi, onAss
         setImmediate: timers.setImmediate,
         clearTimeout: timers.clearTimeout,
         clearInterval: timers.clearInterval,
-        clearImmediate: timers.clearImmediate,
+        clearImmediate: timers.clearImmediate
+    });
 
-        require: (...args) => {
-            const key = args?.[0],
+    const wrappedCode = `
+        ;((originalRequire) => {
```

Comment:
This right now will keep on wrapping require with each execution, but we can move it inside the `initialize` event and only do it once, if the general direction seems satisfactory.